backend-comparison/src/persistence/base.rs

```diff
@@ -1,100 +1,105 @@
-use std::{
-    collections::HashMap,
-    fs::{create_dir_all, File},
-    path::PathBuf,
-    time::Duration,
-};
+use std::fs;
 
-use burn::tensor::backend::Backend;
+use burn::{
+    serde::{ser::SerializeStruct, Serialize, Serializer},
+    tensor::backend::Backend,
+};
 use burn_common::benchmark::BenchmarkResult;
 use dirs;
 use serde_json;
-
-type BenchmarkCommitResults = HashMap<String, BenchmarkOpResults>;
-type BenchmarkOpResults = HashMap<String, BenchmarkBackendResults>;
-type BenchmarkBackendResults = HashMap<String, StampedBenchmarks>;
-type StampedBenchmarks = HashMap<u128, Vec<Duration>>;
+use uuid::Uuid;
 
 #[derive(Default)]
-pub struct Persistence {
-    results: BenchmarkCommitResults,
+pub struct BenchmarkRecord {
+    backend: String,
+    device: String,
+    results: BenchmarkResult,
 }
 
-impl Persistence {
-    /// Updates the cached backend comparison json file with new benchmarks results.
-    ///
-    /// The file has the following structure:
-    ///
-    ///  {
-    ///    "GIT_COMMIT_HASH":
-    ///      {
-    ///        "BENCHMARK_NAME (OP + SHAPE)": {
-    ///          "BACKEND_NAME-DEVICE": {
-    ///            "TIMESTAMP": \[
-    ///              DURATIONS
-    ///           \]
-    ///         }
-    ///       }
-    ///    }
-    ///  }
-    pub fn persist<B: Backend>(benches: Vec<BenchmarkResult>, device: &B::Device) {
-        for bench in benches.iter() {
-            println!("{}", bench);
-        }
-        let cache_file = dirs::home_dir()
-            .expect("Could not get home directory")
-            .join(".cache")
-            .join("backend-comparison")
-            .join("db.json");
+/// Save the benchmarks results on disk.
+///
+/// The structure is flat so that it can be easily queried from a database
+/// like MongoDB.
+///
+/// ```txt
+///  [
+///    {
+///      "backend": "backend name",
+///      "device": "device name",
+///      "git_hash": "hash",
+///      "name": "benchmark name",
+///      "operation": "operation name",
+///      "shapes": ["shape dimension", "shape dimension", ...],
+///      "timestamp": "timestamp",
+///      "numSamples": "number of samples",
+///      "min": "duration in seconds",
+///      "max": "duration in seconds",
+///      "median": "duration in seconds",
+///      "mean": "duration in seconds",
+///      "variance": "duration in seconds"
+///      "rawDurations": ["duration 1", "duration 2", ...],
```

Comment:
`rawDurations` => `raw_durations`

Reply:
This is on purpose, I have a comment on it in the `seralize` function. This is the convention for MongoDB field names.